### PR TITLE
Change LocationSerializer

### LINKED ISSUE #50: Change Location `geopoint` input
- [ ] Change Location serializer to take input of `CSV`
- [ ] Use `CSV` data to populate the `geopoint` field
- [ ] Add Serializer tests
- [ ] Change Locaiton ViewSet and its tests to reflect the above

### PR DESCRIPTION
**Pull request by DavisRayM**
Fix #50

### CODE REVIEW

File: tasking/serializers/location.py [hunk 1]
```diff
@@ -16,11 +17,25 @@ class LocationSerializer(serializers.ModelSerializer):
     Location serializer class
     """
 
+    def to_internal_value(self, data):
```
Comment (moshthepitt) [inline tasking/serializers/location.py]: This should not be in the serializer in my view.  Needs to be a custom field, like so:

```python
class GeoPointField(serializers.Field):
    """
    Custom field for GeoPoint
    """

    def to_internal_value(self, data):
        """
        docstring goes here
        """
        geopoint = data.get('geopoint')

        if geopoint is None:
            self.fail('some error message')

        geopoint_split = geopoint.split(',')
        lon = int(geopoint_split[0])
        lat = int(geopoint_split[1])

        return Point(lon, lat)
```

Comment (moshthepitt) [reply tasking/serializers/location.py]: Please have a look at: http://www.django-rest-framework.org/api-guide/fields/#examples

Comment (moshthepitt) [reply tasking/serializers/location.py]: Also try to not use `long` as a var name.  In certain versions of Python it is a reserved work

File: tests/viewsets/test_locations.py [hunk 2]
```diff
@@ -100,13 +100,16 @@ def test_create_with_bad_data(self):
...
+            geopoint='30,10',
+            shapefile=mocked_location_with_shapefile.shapefile,
+            )
 
         view2 = LocationViewSet.as_view({'post': 'create'})
         request2 = self.factory.post('/locations', data_shapefile)
         # Need authenticated user
         force_authenticate(request2, user=alice_user)
+        # make request mutable
+        request2.POST._mutable = True
```
Comment (moshthepitt) [inline tests/viewsets/test_locations.py]: This seems too weird for me.  Let us find a way that does not need us to do this, OR give sufficient reasons why we cant find such a way.

My reasoning is that in the case of request.POST, it seems no activity in the server side should need to alter the request's data. And hence immutable objects are more suited, not to mention they have substantial performance advantages.

File: tasking/serializers/location.py [hunk 3]
```diff
@@ -5,22 +5,43 @@
...
+        Custom conversion for Geopoint field
+        """
+        geopoint = data
+
+        if geopoint is not None:
+            geopoint_split = geopoint.split(',')
+            lon = int(geopoint_split[0])
+            lat = int(geopoint_split[1])
+
+        return Point(lon, lat)
```
Comment (moshthepitt) [inline tasking/serializers/location.py]: Can this ever raise an exception, for example when lon and lat cannot be valid for a Point?  Can we handle such exceptions and add tests to show that we handle them?

File: tests/viewsets/test_locations.py (hunk 2 above)
Comment (DavisRayM) [reply tests/viewsets/test_locations.py]: Found an alternative way that doesn't require us setting the Post Mutability to True. Used it in the latest commit.

Simple changed the definitions of the data from data = dict() to data = {}

File: tasking/serializers/location.py (hunk 3 above)
Comment (DavisRayM) [reply tasking/serializers/location.py]: Tested it out. There is no case at which this can raise an exception.

**Pull request merged**